dotCMS/src/main/java/com/dotcms/graphql/business/ContentAPIGraphQLFieldsProvider.java

```diff
@@ -0,0 +1,132 @@
+
+        List<GraphQLFieldDefinition> fieldDefinitions = new ArrayList<>();
+
+        contentTypeList.forEach((type) -> {
+            fieldDefinitions.add(createCollectionField(type));
+        });
+
+        // Each BaseType as query'able collection field
+        InterfaceType.valuesAsSet().forEach((type)->{
+            fieldDefinitions.add(createCollectionFieldForBaseType(type));
+        });
+
+        CollectionUtils.filter(fieldDefinitions, PredicateUtils.notNullPredicate());
+        return fieldDefinitions;
+    }
+
+    /**
+     * Given a {@link ContentType} it creates a {@link GraphQLFieldDefinition} that represents a queryable collection for the type
+     * The name of the field will follow the convention "{typeVariable}Collection".
+     * <p>
+     * E.g Content Type Variable: Product. Resulting field name: ProductCollection
+     * @param type the type we want to build a collection for
+     * @return the field definition representing the collection
+     */
+
+    private GraphQLFieldDefinition createCollectionField(ContentType type) {
+        try {
+            return newFieldDefinition()
+                    .name(TypeUtil.collectionizedName(type.variable()))
+                    .argument(GraphQLArgument.newArgument()
+                            .name("query")
+                            .type(GraphQLString)
+                            .build())
+                    .argument(GraphQLArgument.newArgument()
+                            .name("limit")
+                            .type(GraphQLInt)
+                            .build())
+                    .argument(GraphQLArgument.newArgument()
+                            .name("offset")
+                            .type(GraphQLInt)
+                            .build())
+                    .argument(GraphQLArgument.newArgument()
+                            .name("sortBy")
+                            .type(GraphQLString)
+                            .build())
+                    .type(list((ContentAPIGraphQLTypesProvider.
+                            INSTANCE.getCachedTypesAsMap().get(type.variable()))))
+                    .dataFetcher(new ContentletDataFetcher()).build();
+        } catch (DotDataException e) {
+            Logger.error("Unable to generate Collection for type: " + type.variable(), e);
+        }
+        return null;
+    }
+
+    /**
+     * Given a {@link GraphQLType} it creates a {@link GraphQLFieldDefinition} that represents a queryable collection for the type
+     * The name of the field will follow the convention "{typeName}Collection".
+     * <p>
+     * E.g Type Name: WidgetBaseType. Resulting field name: WidgetBaseTypeCollection
+     * <p>
+     * This method is meant to be used for BaseTypes
+     *
+     * @param type the type we want to build a collection for
+     * @return the field definition representing the collection
+     */
+
+    private GraphQLFieldDefinition createCollectionFieldForBaseType(GraphQLInterfaceType type) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'type' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5642169)